oeps/oep-0052-arch-event-bus-technology.rst

```diff
@@ -0,0 +1,264 @@
+==============================
+OEP-0052: Event Bus Technology
+==============================
+
+.. This OEP template is based on Nygard's Architecture Decision Records.
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP-52
+     - :doc:`OEP-0052 <oep-0052-arch-event-bus-technology>`
+   * - Title
+     - Event Bus Technology
+   * - Last Modified
+     - 2021-08-26
+   * - Authors
+     - Feanil Patel <feanil@edx.org>, Robert Raposa <rraposa@edx.org>
+   * - Arbiter
+     - TBD
+   * - Status
+     - Draft
+   * - Type
+     - Architecture
+   * - Created
+     - 2021-08-16
+   * - Review Period
+     - TBD
+
+Overview
+--------
+
+* Adding an event bus to the Open edX platform allows asynchronous event messaging across services, which enables a number of improvements aligned with our architectural goals.
+
+* We are trialing Kafka to implement an Open edX event bus. This decision will be updated as we commit or change direction.
```

Comment:
Whatever the event bus ends up being, please take into account the deployment costs. In my experience, Kafka is a great piece of software that is very difficult to deploy reliably. For reference, here is the complete list of configuration parameters: https://kafka.apache.org/documentation/#configuration Note that Zookeeper needs to be configured separately.
I would strongly suggest a benchmark of Redis Streams: https://redis.io/topics/streams-intro My personal experience with it has been extremely positive. I think it should cover all of the classic use cases.